pkg/endpoints/pipeline.go

```diff
@@ -403,8 +555,42 @@ func getDateTimeAsString() string {
 	return strconv.FormatInt(time.Now().Unix(), 10)
 }
 
+// defines and creates a resource of a specifed type and returns a pipeline resource reference for this resource
+// takes a manual pipeline run data struct, namespace for the resource creation, resource name to refer to it and the resource type
+func (r Resource) createPipelineResource(resourceData ManualPipelineRun, namespace, resourceName string, resourceType v1alpha1.PipelineResourceType) v1alpha1.PipelineResourceRef {
+	logging.Log.Debug("Creating PipelineResource of type %s", resourceType)
+
+	startTime := getDateTimeAsString()
+
+	registryURL := resourceData.REGISTRYLOCATION
+	resourceName = fmt.Sprintf("%s-%s", resourceName, startTime)
+
+	var paramsForResource []v1alpha1.Param
+	var typeToUse v1alpha1.PipelineResourceType
+	// Unique names are required so timestamp them.
+	if resourceType == v1alpha1.PipelineResourceTypeGit {
+		typeToUse = resourceType
+		paramsForResource = []v1alpha1.Param{{Name: "revision", Value: resourceData.GITCOMMIT}, {Name: "url", Value: resourceData.REPOURL}}
+	} else if resourceType == v1alpha1.PipelineResourceTypeImage {
+		urlToUse := fmt.Sprintf("%s/%s:%s", registryURL, strings.ToLower(resourceData.REPONAME), resourceData.GITCOMMIT)
+		typeToUse = resourceType
+		paramsForResource = []v1alpha1.Param{{Name: "url", Value: urlToUse}}
+	}
+
+	pipelineResource := definePipelineResource(resourceName, namespace, paramsForResource, typeToUse)
+	createdPipelineImageResource, err := r.PipelineClient.TektonV1alpha1().PipelineResources(namespace).Create(pipelineResource)
+	if err != nil {
+		logging.Log.Errorf("could not create pipeline %s resource to be used in the pipeline, error: %s", resourceType, err)
```

Comment:
Please return an error here. The function should return an error if it fails to create the PipelineResource.